src/components/List.js

```diff
@@ -36,29 +45,49 @@ const List = ({ token }) => {
     <div>
       <h1>List</h1>
 
-      {docs.length === 0 && (
+      {docs.length === 0 ? (
         <section>
           <p>Your shopping list is currently empty.</p>
           <a href="/add-item">Add an Item</a>
         </section>
+      ) : (
+        <div>
+          <label htmlFor="search-bar">Filter Items</label>
+          <br />
+          <input
+            type="text"
+            name="search-bar"
+            id="search-bar"
+            placeholder="Start typing here..."
+            value={searchInput}
+            onChange={handleSearchChange}
+          />
+          <input type="reset" onClick={handleClear} />
+        </div>
       )}
 
       {errorMessage && <Error errorMessage={errorMessage} />}
 
       <ul style={{ listStyleType: 'none' }}>
         {docs &&
-          docs.map((doc) => (
-            <li key={doc.id}>
-              <input
-                type="checkbox"
-                name={doc.itemName}
-                id={doc.id}
-                onChange={handleCheckbox}
-                checked={checkPurchasedDate(doc.lastPurchased)}
-              />
-              {doc.itemName}
-            </li>
-          ))}
+          docs
```

Comment:
One thing that it's always good to be vigilantly on guard for, especially for data coming from an API that we don't have much control over, is null-pointer errors. Any time that there's a bunch of chained dot-notation, there's always an increased risk of this. 

In the past, we would have to use phrases like `doc && doc.itemName && doc.itemName.toLowerCase()...` or rely on third party utilities like lodash get (https://lodash.com/docs/4.17.15#get) in order to mitigate this risk.

Luckily, the brand new JS feature of optional-chaining (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) allows us to super easily add null-pointer error protection into our code. Even more luckily, it appears our project appears to be set up to use this modern JS feature!

If we were to use optional chaining here, I think it would look something like:

```javascript
docs
          ?.filter((doc) =>
            doc?.itemName
              ?.toLowerCase()
              ?.includes(searchInput?.toLowerCase()?.trim?.()),
          )
          ?.map(...
```
(note, I don't use optional chaining much b/c my project at work doesn't support it, so definitely triple-check my suggestion)

Reply:
Thanks for sharing @MikeBLambert. 
I find it interesting that optional-chaining isn't supported by Samsung Internet which is the official browser for Samsung devices (though apparently they are only [2.52%](https://caniuse.com/?search=optional%20chaining%20operator%20%3F) of global users), so if we were releasing the app for widespread use that might be something we would want to consider.

Reply:
Absolutely, that is totally the kind of thing that you'd want to consider and I love that you brought that consideration up. In fact, I believe that long, long ago (or at least some time before the three years ago when I became a developer...) a lot of time and effort was spent getting code to work with different browsers.

Lucky for us though, our project uses Babel, which takes care of translating our fancy, modern JS into boring, old, "I remember when I used to walk to school in the snow, uphill, both ways..." JS.

If you open up the node_modules, you can see that there is a dependency called @babel/plugin-proposal-optional-chaining. This is the package that specifically deals with optional-chaining.
![image](https://user-images.githubusercontent.com/34964885/106984664-ded24700-671c-11eb-8faa-485ffe9f3b29.png)

Now, if you look towards the bottom of our package.json, you'll see that there's a section called "browserlist". I believe that this is what Babel references when it does this translation. (I'm pretty fuzzy on a lot of this stuff though, so I wouldn't be surprised or insulted if I was corrected on the details...) Interestingly enough, in looking at [browserlist](https://www.npmjs.com/package/browserslist#browsers), it appears that they actually reference the site that you linked to in your reply. 🙌 🙌

Reply:
Oh right, how could I forget about the "magic" of Babel 🧙